aten/src/ATen/core/thread_pool.cpp

```diff
@@ -117,8 +117,18 @@ void ThreadPool::main_loop(std::size_t index) {
   } // while running_
 }
 
+c10::optional<size_t> num_threads(c10::optional<size_t> val) {
+  static c10::optional<size_t> n_threads = val;
```

Comment:
so what happens if someone calls this function first to get num threads? it'd be none forever? imho, it's better to default to 1 here there in that case and remove the conditional in ThreadPool.